packages/core/src/execution-engine/workflow-execute.ts

```diff
@@ -1089,72 +1089,68 @@ export class WorkflowExecute {
 		return customOperation;
 	}
 
-	/** Executes the given node */
-	// eslint-disable-next-line complexity
-	async runNode(
-		workflow: Workflow,
-		executionData: IExecuteData,
-		runExecutionData: IRunExecutionData,
-		runIndex: number,
-		additionalData: IWorkflowExecuteAdditionalData,
-		mode: WorkflowExecuteMode,
-		abortSignal?: AbortSignal,
-	): Promise<IRunNodeResponse> {
-		const { node } = executionData;
-		let inputData = executionData.data;
-
-		if (node.disabled === true) {
-			// If node is disabled simply pass the data through
-			// return NodeRunHelpers.
-			if (inputData.hasOwnProperty('main') && inputData.main.length > 0) {
-				// If the node is disabled simply return the data from the first main input
-				if (inputData.main[0] === null) {
-					return { data: undefined };
-				}
-				return { data: [inputData.main[0]] };
+	/**
+	 * Handles execution of disabled nodes by passing through input data
+	 */
+	private handleDisabledNode(inputData: ITaskDataConnections): IRunNodeResponse {
+		if (inputData.hasOwnProperty('main') && inputData.main.length > 0) {
+			// If the node is disabled simply return the data from the first main input
+			if (inputData.main[0] === null) {
+				return { data: undefined };
 			}
-			return { data: undefined };
+			return { data: [inputData.main[0]] };
 		}
+		return { data: undefined };
+	}
 
-		const nodeType = workflow.nodeTypes.getByNameAndVersion(node.type, node.typeVersion);
-
-		const isDeclarativeNode = nodeType.description.requestDefaults !== undefined;
-
-		const customOperation = this.getCustomOperation(node, nodeType);
-
-		let connectionInputData: INodeExecutionData[] = [];
+	/**
+	 * Prepares input data for node execution based on node type and workflow settings
```

Comment:
This docline could use some love. What does it mean to "prepare" input data? The args mentioned are already clear from the signature. Let's either improve or remove?